cli/integration/tests/waiter/test_cli.py

```diff
@@ -2394,12 +2394,15 @@ def __test_signal(self, get_possible_instances_fn, signal_type=None, signal_flag
             self.assertEqual(0, cp.returncode, cp.stderr)
             self.assertIn(f'Successfully sent {signal_type_output} to', stdout)
             kill_fn = lambda insts: (min_instances - 1) == len(insts['active-instances']) and \
-                                    1 == len(insts['killed-instances'])
+                                    1 >= len(insts['killed-instances'])
             util.wait_until_routers_service(self.waiter_url, service_id, lambda service: kill_fn(service['instances']))
-            killed_instances = util.specific_instances_for_service(self.waiter_url, service_id, 'killed-instances')
+            killed_instances = util.specific_instances_for_service(self.waiter_url, service_id,'killed-instances')
             killed_instance_id = killed_instances[0]['id']
-            active_instances = util.specific_instances_for_service(self.waiter_url, service_id,'active-instances' )
-            self.assertNotIn(killed_instance_id, active_instances, "instance is in active instances")
+            active_instances_map = util.specific_instances_for_service(self.waiter_url, service_id,'active-instances')
+            active_instances = []
+            for i in range(len(active_instances_map)):
+                active_instances.append(active_instances_map[i]['id'])
```

Comment:
Please use map to extract the `active_instance_ids` (this name also better reflects that the values are ids).
https://www.geeksforgeeks.org/python-map-function/